### ISSUE TITLE
Static files inaccessible

### ISSUE THREAD

**Issue by Yakumo-Yukari** [ire4ever1190/mike#57]
```

project/
  └── public/
      └── static/
          └── js/
              └── alpine.min.js
```

```
<!DOCTYPE html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script src="/static/js/alpine.min.js"></script>
</head>
<body class="bg-gray-50 min-h-screen flex flex-col">
    <main class="flex-grow">
        {% block content %}{% endblock %}
    </main>
</body>
</html>
```

`servePublic("public/", "/") `

`>nim c -r -d:mikeStaticFiles src/app`

Error:

```
{
    "kind": "NotFoundError",
    "detail": "static/js/alpine.min.js not found",
    "status": 404
}
```

`GET http://127.0.0.1:8080/static/js/alpine.min.js net::ERR_ABORTED 404 (Not Found)`

**Comment by Yakumo-Yukari** [comment]
servePublic("src/public", "/", staticFiles = true)  No access to files in the public directory
servePublic("src/public/static/js", "/static/js", staticFiles = true) This is the only way to access the js file 

How to solve the problem that servePublic(‘src/public’, ‘/’, staticFiles = true) can't access public directory files?

**Comment by ire4ever1190** [comment]
I can't seem to replicate, what OS and version of Nim are you using?
Setup that I've done, might've done something wrong

`src/main.nim`
```nim
import mike
import mike/public

servePublic("src/public/", "/", staticFiles=true)

run()
```
`src/public/static/js/alpine.min.js`
```js
console.log("test")
```

Then running `nim c -r src/main` and `curl http://127.0.0.1:8080/static/js/alpine.min.js` I get 
```
console.log("test")
```

**Comment by Yakumo-Yukari** [comment]
> I can't seem to replicate, what OS and version of Nim are you using? Setup that I've done, might've done something wrong
> 
> `src/main.nim`
> 
> ```nim
> import mike
> import mike/public
> 
> servePublic("src/public/", "/", staticFiles=true)
> 
> run()
> ```
> 
> `src/public/static/js/alpine.min.js`
> 
> ```js
> console.log("test")
> ```
> 
> Then running `nim c -r src/main` and `curl http://127.0.0.1:8080/static/js/alpine.min.js` I get
> 
> ```
> console.log("test")
> ```

OS:
`Windows11`

Nim:
`2.2.0`

GCC
```
> gcc --version
gcc.exe (GCC) 14.2.0
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

CC: wepoll

![1](https://github.com/user-attachments/assets/d4f9f29e-0dee-43b0-8a98-ccf071858b38)
![2](https://github.com/user-attachments/assets/36a2c570-bb44-48de-a21e-5dbd3958103d)

![3](https://github.com/user-attachments/assets/6269480f-3bcb-40b9-881c-9fb1f41103de)



```
import 
  mike,
  ../View/homeView

proc routes*() =

  "/" -> get:
    homeView(ctx)

```



> curl http://127.0.0.1:8080/static/js/alpine.min.js
{"kind":"NotFoundError","detail":"static/js/alpine.min.js not found","status":404}

```
```
Error while handling: GET /static/js/alpine.min.js
static/js/alpine.min.js not found
D:\Dev\project\NimProject\NimExample\AtelierRyza\src\app.nim(13) app
C:\Users\.nimble\pkgs2\mike-1.3.3-c6bb4e36b463e956150b2f1a36f85b747cb4ac30\mike\dsl.nim(221) run
C:\Users\.nimble\pkgs2\httpx-0.3.8-31ef637d4136a2c7deebce51c02c27145692d4f5\httpx.nim(684) run
C:\Users\.nimble\pkgs2\httpx-0.3.8-31ef637d4136a2c7deebce51c02c27145692d4f5\httpx.nim(585) eventLoop
C:\Users\.nimble\pkgs2\httpx-0.3.8-31ef637d4136a2c7deebce51c02c27145692d4f5\httpx.nim(478) processEvents
C:\Users\.nimble\pkgs2\mike-1.3.3-c6bb4e36b463e956150b2f1a36f85b747cb4ac30\mike\dsl.nim(170) onRequest (Async)
C:\Users\.nimble\pkgs2\mike-1.3.3-c6bb4e36b463e956150b2f1a36f85b747cb4ac30\mike\public.nim(88) /^file (Async)
```

**Comment by Yakumo-Yukari** [comment]
I'm not sure if this is caused by the system, I've only tested on windows

**Comment by ire4ever1190** [comment]
Thanks for that info!
Can reproduce on windows, looking into a fix 

Turns out I don't handle windows paths correctly 🤦‍♂️

**Comment by ire4ever1190** [comment]
Fixed in `1.3.5`
